### PR TITLE
Added the azure cost-over-usage

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [x] dependencies

## Description

Added the Azure cost over usage functionality.
Created the abstract class for cost over usage

### CODE REVIEW

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/cost_over_usage.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    def __init__(self):
+        super().__init__()
+        self.__cost_mgmt_operations = CostManagementOperations()
+        self.__compute_operations = ComputeOperations()
+        self._subscription_id = self._environment_variables_dict.get('AZURE_SUBSCRIPTION_ID')
+        self.__scope = f'subscriptions/{self._subscription_id}'
+
+    def _verify_active_resources(self, tag_name: str, tag_value: str) -> bool:
+        """
+        This method verify any active virtual instances in all regions by tag_name, tag_value
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/cost_over_usage.py:21]: verifies

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/cost_over_usage.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        """
+        virtual_machines = self.__compute_operations.get_all_instances()
+        for virtual_machine in virtual_machines:
+            tags = virtual_machine.tags
+            user = self.__compute_operations.check_tag_name(tags=tags, tag_name=tag_name)
+            if string_equal_ignore_case(user, tag_value):
+                return True
+        return False
+
+    def _get_cost_based_on_tag(self, start_date: str, end_date: str, tag_name: str, extra_filters: any = None,
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/cost_over_usage.py:37]: Missing document

File: cloud_governance/cloud_resource_orchestration/clouds/common/abstract_cost_over_usage.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
+import logging
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/common/abstract_cost_over_usage.py:1]: Is it a new abstract class that serve AWS and Azure ?

Comment (athiruma) [reply cloud_governance/cloud_resource_orchestration/clouds/common/abstract_cost_over_usage.py:1]: yes

**Pull request merged**